File: Semestrul 3/MAP/lab 10 MAP/model/SortingTask.cs
using seminar10.factory;

namespace seminar10.model;

public class SortingTask : Task
{
    private readonly SortingStrategy _strategy;
    private readonly AbstractSorter _sorter;
    private readonly int[] _vector;

    public SortingTask(string id, string description, SortingStrategy strategy, int[] vector)
        : base(id, description)
    {
        _strategy = strategy;
        _vector = vector;
        _sorter = _strategy == SortingStrategy.Bubble 
            ? new BubbleSort() 
            : new QuickSort();
    }

    public override void Execute()
    {
        _sorter.Sort(_vector);
        foreach (var i in _vector)
        {
            Console.Write(i + " ");
        }
    }
}

File: Semestrul 3/MAP/lab 2 MAP/model/SortingTask.java
package sem1_2.model;

import sem1_2.factory.SortingStrategy;

public class SortingTask extends Task{


    public SortingStrategy strategy;
    public AbstractSorter sorter;
    public int[] vector;

    public SortingTask(String id, String description, SortingStrategy strategy, int[] vector) {
        super(id, description);
        this.strategy = strategy;
        this.vector = vector;
        if (strategy == SortingStrategy.BUBBLE) {
            sorter = new BubbleSort();
        }
        else {
            sorter = new QuickSort();
        }
    }

    @Override
    public void execute(){
        sorter.sort(vector);
        for (int j : vector) {
            System.out.print(j + " ");
        }
    }
}
